app/services/movie_service.rb

```diff
@@ -0,0 +1,26 @@
+class MovieService
+  def top_rated
+    movies = []
+    i = 0
+    while movies.size < 40
+      i +=1
+      movies << to_json("movie/top_rated?page=#{i}")[:results]
+      movies.flatten!
+      break if movies == [nil]
+    end
+    movies
+  end
```

Comment:
```suggestion
  def top_rated
    movies = []
    page = 0
    while movies.size < 40
      page +=1
      movies << to_json("movie/top_rated?page=#{page}")[:results]
      movies.flatten!
      break if movies == [nil]
    end
    movies
  end
```

I think this improve the clarity of this method!